functions.php

```diff
@@ -249,3 +250,127 @@ function ( array $editor_settings, WP_Block_Editor_Context $block_editor_context
 	10,
 	2
 );
+
+/**
+ * I'll move this somewhere else in master theme.
+ *
+ * @return void
+ */
+function register_more_blocks() {
+	register_block_type(
+		'p4/reading-time',
+		[
+			'render_callback' => [ Post::class, 'reading_time_block' ],
+			'uses_context'    => [ 'postId' ],
+		]
+	);
+	register_block_type(
+		'p4/post-author-name',
+		[
+			'render_callback' => function ( array $attributes, $content, $block ) {
+				$author_override = get_post_meta( $block->context['postId'], 'p4_author_override', true );
+				$post_author_id  = get_post_field( 'post_author', $block->context['postId'] );
+
+				$is_override = ! empty( $author_override );
+
+				$name = $is_override ? $author_override : get_the_author_meta( 'display_name', $post_author_id );
+				$link = $is_override ? '#' : get_author_posts_url( $post_author_id );
+
+				$tag_content = $author_override ? $name : "<a href='$link'>$name</a>";
```

Comment:
Shouldn't this variable be called `$author_name` or something similar, rather than `$tag_content`? 🤔

Reply:
No, it's intentional. I mean "tag content" in the sense that the HTML tag's content will be either a link, or just text.

Reply:
Changed it to `$block_content`